invenio_rdm_records/resources/serializers.py

```diff
@@ -21,87 +20,60 @@ class UIJSONSerializer(JSONSerializer):
             vocabulary_name).dump_options()
         if type(vocabulary) is dict:
-            ui_options[top_bucket_key] = {}
+            flattened_vocabulary = {}
             for key in vocabulary.keys():
                 for option in vocabulary[key]:
-                    ui_options[top_bucket_key][option['value']] = \
-                        str(option['text'])
+                    flattened_vocabulary[option['value']] = option
+
         else:
             for option in vocabulary:
-                ui_options[option['value']] = str(option['text'])
-        return ui_options
+                flattened_vocabulary[option['value']] = option
+        return flattened_vocabulary
 
-    def _set_bucket_label(self, labels_map, bucket):
-        """Set bucket label from config.
-
-        :params labels_map: Dict of transforming bucket values to labels.
-        :params bucket: Current aggregation bucket.
-        """
-        bucket_label = labels_map.get(bucket["key"])
-        if bucket_label:
-            bucket["label"] = bucket_label
-        if bucket.get("subtype"):
-            bucket["subtype"]["buckets"] = list(map(
-                partial(self._set_bucket_label, labels_map),
-                bucket["subtype"]["buckets"]))
-        return bucket
-
-    def _set_buckets_labels(self, label_map, buckets):
-        """Map buckets values to labels according to configuration.
-
-        :params labels_map: Dict of transforming bucket values to labels.
-        :params buckets: Current aggregation bucket list.
-        """
-        return list(map(
-                partial(self._set_bucket_label, label_map),
-                buckets))
-
-    def _serialize_agg(self, agg_obj, labels_map):
-        """Inject ui config for aggregation.
+    def _enrich_buckets_with_labels(self, agg_obj, vocabulary):
+        """Inject labels in the aggregation buckets.
 
         :params agg_obj: Current aggregation object.
-        :params labels_map: Dict of transforming bucket values to labels.
+        :params vocabulary: Dict with vocabulary data.
         """
-        if agg_obj:
-            agg_obj["buckets"] = self._set_buckets_labels(
-                labels_map, agg_obj["buckets"]
-            )
-
-    def _serialize_resource_type_agg(self, resource_type_agg):
-        """Inject ui config for `resource_type` aggregation."""
-        resource_type_labels = self._serialize_ui_options_from_vocabulary(
-            "resource_type", "type")
-        self._serialize_agg(resource_type_agg, resource_type_labels["type"])
-
-    def _serialize_access_right_agg(self, access_right_agg):
-        """Inject ui config for `access_right` aggregation."""
-        access_right_labels = self._serialize_ui_options_from_vocabulary(
-            "access_right")
-        self._serialize_agg(access_right_agg, access_right_labels)
+        buckets = agg_obj['buckets']
+        for bucket in buckets:
+            for k in list(bucket):
```

Comment:
I am confused here....buckets is an array of objects as far as I remember....why are you iterating through it???